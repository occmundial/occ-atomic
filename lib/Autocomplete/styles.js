import idx from 'idx';
import { createUseStyles } from 'react-jss';

import spacing from '../subatomic/spacing';

export default createUseStyles({
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: props => idx(props, _ => _.textfieldProps.label) || idx(props, _ => _.textfieldProps.lockHeight) ? spacing.xLarge : 40,
        left: 0,
        width: '100%'
    }
});
