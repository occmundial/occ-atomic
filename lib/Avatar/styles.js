import { createUseStyles } from 'react-jss';

export default createUseStyles({
    circle: {
        position:'relative',
        width: props => props.size ? props.size : 70,
        height: props => props.size ? props.size : 70,
        borderRadius:'50%',
        overflow:'hidden'
    }
});
