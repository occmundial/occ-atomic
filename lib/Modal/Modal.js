import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card';
import Icon from '../Icon';
import Text from '../Text';
import Flexbox from '../Flexbox';
import Button from '../Button';
import colors from '../subatomic/colors';
import grid from '../subatomic/grid';

/**
* The Modal component is rendered in any given container, avoiding z-index an position conflicts. Your application must have the logic to show or hide the Modal component.
*/
class Modal extends React.Component {

    constructor(props) {
        super(props);
        this.state = { mobile: false };
        this.onKeyDown = this.onKeyDown.bind(this);
    }

    componentDidMount() {
        window.addEventListener('keydown', this.onKeyDown);
        window.addEventListener('resize', this.resize);
        this.resize();
    }

    componentWillUnmount() {
        window.removeEventListener('keydown', this.onKeyDown);
        window.removeEventListener('resize', this.resize);
    }

    resize = () => {
        this.setState({ mobile: window.innerWidth < grid.xs });
    }

    onKeyDown({ which }) {
        const { onClose } = this.props;
        if(onClose && which == 27) onClose();
    }

    avoidClose(e) {
        e.stopPropagation();
    }

    render() {
        const { mobile } = this.state;
        const { classes, children, onClose, title, mainBtn, show, secBtn, size, imgTop, imgLeft, onTransitionEnd, fullSize } = this.props;

        const closeButton = (
            <div className={classes.closeIcon}>
                <Icon iconName="close" width={24} height={24} colors={[colors.grey900]} onClick={onClose} />
            </div>
        );

        return (
            <div className={`${classes.overlay} ${show ? classes.overlayShow : classes.overlayHide}${!onClose ? ` ${classes.noClose}` : ''}`} onClick={onClose} onTransitionEnd={onTransitionEnd}>
                <div className={classes.cardWrapper}>
                    <div className={classes.cardBlock} onClick={this.avoidClose}>
                        <Card shadow={5} className={`${classes.card}${size ? ` ${classes[size]}` : ''}${show ? ` ${classes.cardShow}` : ` ${classes.cardHide}`}`}>
                            <Flexbox display="flex" direction={imgLeft.img ? 'row' : 'col'}>
                                {(imgLeft.img || imgTop.img) && (
                                    <div className={imgLeft.img ? classes.imgLeft : classes.imgTop}>
                                        {(imgTop.img && onClose) && (
                                            <div className={classes.closePosition}>
                                                {closeButton}
                                            </div>
                                        )}
                                    </div>
                                )}
                                <Flexbox flex={imgLeft.img ? '1' : null}>
                                    {(fullSize && mobile) ? (
                                        <div className={classes.top}>
                                            <Flexbox alignItems="end" className={classes.top}>
                                                {(!imgTop.img && onClose) && closeButton}
                                            </Flexbox>
                                            {title && (
                                                <Text heading className={classes.title}>{title}</Text>
                                            )}
                                        </div>
                                    ) : (
                                        <Flexbox display="flex" justifyContent="end" alignItems="start" className={classes.top}>
                                            {title && (
                                                <Flexbox flex="1">
                                                    <Text heading className={classes.title}>{title}</Text>
                                                </Flexbox>
                                            )}
                                            {(!imgTop.img && onClose) && closeButton}
                                        </Flexbox>
                                    )}
                                    <div className={classes.content}>
                                        {children}
                                    </div>
                                    {mainBtn && (
                                        <div className={classes.bottom}>
                                            {secBtn && (
                                                <Button
                                                    theme="ghostGrey"
                                                    className={classes.secBtn}
                                                    onClick={secBtn.onClick}
                                                    href={secBtn.href}
                                                    target={secBtn.target}
                                                    loading={secBtn.loading}
                                                    disabled={secBtn.disabled}
                                                    id={secBtn.id}
                                                >
                                                    {secBtn.text}
                                                </Button>)
                                            }
                                            <Button
                                                className={classes.mainBtn}
                                                onClick={mainBtn.onClick}
                                                href={mainBtn.href}
                                                target={mainBtn.target}
                                                loading={mainBtn.loading}
                                                disabled={mainBtn.disabled}
                                                id={mainBtn.id}
                                            >
                                                {mainBtn.text}
                                            </Button>
                                        </div>
                                    )}
                                </Flexbox>
                            </Flexbox>
                        </Card>
                    </div>
                </div>
            </div>
        );
    }
}

Modal.defaultProps = {
    size: 'md',
    imgLeft: {},
    imgTop: {}
};

Modal.propTypes = {
    /** prop */
    classes: PropTypes.object.isRequired,
    children: PropTypes.node,
    /** Boolean value to show or hide the Modal component. */
    show: PropTypes.bool,
    /** DOM Element to append the Modal component into it. If you don't specify a container the Modal will be rendered in the exact same DOM position you used it. This can in some cases generate problems with z-index or position. */
    container: PropTypes.object,
    /** Function to call when the overlay or close icon are clicked or when you press the Esc key. */
    onClose: PropTypes.func,
    /** Size of the Modal. Select any of these: sm, md, lg, xl. */
    size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl']),
    /** Title to display at the top of the Modal. */
    title: PropTypes.string,
    /** Object with props for the main button. It will render a Button with the main theme. */
    mainBtn: PropTypes.shape({
        text: PropTypes.string,
        onClick: PropTypes.func,
        href: PropTypes.string,
        target: PropTypes.string,
        disabled: PropTypes.bool,
        loading: PropTypes.bool
    }),
    /** Object with props for the secondary button. It will render a Button with the ghostGrey theme. (NOTE: If you don't send a mainBtn object the secBtn will not work) */
    secBtn: PropTypes.shape({
        text: PropTypes.string,
        onClick: PropTypes.func,
        href: PropTypes.string,
        target: PropTypes.string,
        disabled: PropTypes.bool,
        loading: PropTypes.bool
    }),
    /** Image url to render it on the top of the modal dialog. */
    imgTop: PropTypes.shape({
        img: PropTypes.string,
        color: PropTypes.string,
        position: PropTypes.oneOf([
            'left', 'center', 'right'
        ]),
        size: PropTypes.oneOf([
            'contain', 'cover'
        ]),
        height: PropTypes.number,
        personalize: PropTypes.bool,
    }),
    /** Image url to render it on the left of the modal dialog. */
    imgLeft: PropTypes.shape({
        img: PropTypes.string,
        color: PropTypes.string,
        position: PropTypes.oneOf([
            'top', 'center', 'bottom'
        ]),
        size: PropTypes.oneOf([
            'contain', 'cover'
        ])
    }),
    /** This function is used by an internal wrapper component to manage the hiding animation and unmounting properly. */
    onTransitionEnd: PropTypes.func,
    /** Boolean value to make full size the Modal component in mobile size. */
    fullSize: PropTypes.bool
};

export default Modal;
