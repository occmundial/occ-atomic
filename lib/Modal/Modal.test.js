import React from 'react';
import { shallow, mount } from 'enzyme';
import Modal from './Modal';
import styles from './styles';
import ModalPortal from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Modal", () => {

    it('matches the snapshot', () => {
        const onClose = jest.fn();
        const wrapper = shallow(
            <Modal classes={classes} onClose={onClose} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('calls the onClose function when clicking the overlay', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal classes={classes} onClose={onClose} />
        );
        wrapper.find('.overlay').simulate('click');
        expect(onClose.mock.calls.length).toBe(1);
    });

    it('calls the onClose function when clicking the close icon', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal classes={classes} onClose={onClose} testId="modal-test" />
        );
        wrapper.find({ "data-testid": 'modal-test__close-icon'}).simulate('click');
        expect(onClose.mock.calls.length).toBe(1);
    });

    it('calls the onClose function when pressing the Esc key', () => {
        const onClose = jest.fn();
        const map = {};
        window.addEventListener = jest.fn((event, cb) => {
            map[event] = cb;
        });
        mount(
            <Modal classes={classes} onClose={onClose} />
        );
        map.keydown({ which: 27 });
        expect(onClose.mock.calls.length).toBe(1);
    });

    it('toggles between show and hide', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal classes={classes} onClose={onClose} show={true} />
        );
        expect(wrapper.find('.show').length).toBe(1);
        wrapper.setProps({ show: false });
        expect(wrapper.find('.hide').length).toBe(1);
    });

    it('renders the title', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal classes={classes} onClose={onClose} title="Modal title" />
        );
        expect(wrapper.find('.title').at(0).text()).toBe('Modal title');
    });

    it('renders the content', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal classes={classes} onClose={onClose}>
                Content
            </Modal>
        );
        expect(wrapper.find('.content').at(0).text()).toBe('Content');
    });

    it('renders functional main and secondary buttons', () => {
        const onClose = jest.fn();
        const wrapper = mount(
            <Modal
                classes={classes}
                onClose={onClose}
                mainBtn={{
                    text:'Main button',
                    onClick: onClose
                }}
                secBtn={{
                    text:'Secondary button',
                    onClick: onClose
                }}
                testId="modal"
            />
        );
        const mainBtn = wrapper.find({ 'data-testid': 'modal__button-main' }).at(0);
        const secBtn = wrapper.find({ 'data-testid': 'modal__button-secondary' }).at(0);
        mainBtn.simulate('click');
        secBtn.simulate('click');
        expect(mainBtn.text()).toBe('Main button');
        expect(secBtn.text()).toBe('Secondary button');
        expect(onClose.mock.calls.length).toBe(2);
    });

});

describe("ModalPortal", () => {

    it('matches the snapshot', () => {
        const modalRoot = global.document.createElement('div');
        const wrapper = shallow(
            <ModalPortal container={modalRoot} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('mounts and unmounts the component', () => {
        const onClose = jest.fn();
        const modalRoot = global.document.createElement('div');
        modalRoot.setAttribute('id', 'modal-root');
        const body = global.document.querySelector('body');
        body.appendChild(modalRoot);
        const wrapper = mount(
            <ModalPortal container={modalRoot} onClose={onClose} />
        );
        expect(modalRoot.hasChildNodes()).toBeFalsy();
        wrapper.setProps({ show: true });
        expect(modalRoot.hasChildNodes()).toBeTruthy();
        wrapper.setProps({ show: false });
        wrapper.setState({ mount: false });
        expect(modalRoot.hasChildNodes()).toBeFalsy();
    });

});

describe("Modal styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});
