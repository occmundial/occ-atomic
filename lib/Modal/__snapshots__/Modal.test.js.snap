// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    classes={
      Object {
        "@global": "@global",
        "@keyframes modalFadeDown": "@keyframes modalFadeDown",
        "@keyframes modalFadeIn": "@keyframes modalFadeIn",
        "bottom": "bottom",
        "card": "card",
        "cardBlock": "cardBlock",
        "cardHide": "cardHide",
        "cardShow": "cardShow",
        "cardWrapper": "cardWrapper",
        "closeIcon": "closeIcon",
        "closePosition": "closePosition",
        "content": "content",
        "imgLeft": "imgLeft",
        "imgTop": "imgTop",
        "lg": "lg",
        "mainBtn": "mainBtn",
        "md": "md",
        "noClose": "noClose",
        "overlay": "overlay",
        "overlayHide": "overlayHide",
        "overlayShow": "overlayShow",
        "secBtn": "secBtn",
        "sm": "sm",
        "top": "top",
        "xl": "xl",
      }
    }
    onClose={[MockFunction]}
    size="md"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cardWrapper"
      >
        <div
          className="cardBlock"
          onClick={[Function]}
        >
          <Jss(Card)
            className="card md cardHide"
            shadow={5}
          >
            <Jss(Flexbox)
              direction="col"
              display="flex"
            >
              <Jss(Flexbox)
                flex="1"
              >
                <Jss(Flexbox)
                  alignItems="start"
                  className="top"
                  display="flex"
                  justifyContent="end"
                >
                  <div
                    className="closeIcon"
                  >
                    <Jss(Icon)
                      colors={
                        Array [
                          "#222222",
                        ]
                      }
                      height={24}
                      iconName="close"
                      onClick={[MockFunction]}
                      transition="0.3s all"
                      width={24}
                    />
                  </div>
                </Jss(Flexbox)>
                <div
                  className="content"
                />
              </Jss(Flexbox)>
            </Jss(Flexbox)>
          </Jss(Card)>
        </div>
      </div>,
      "className": "overlay overlayHide",
      "onClick": [MockFunction],
      "onTransitionEnd": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cardBlock"
          onClick={[Function]}
        >
          <Jss(Card)
            className="card md cardHide"
            shadow={5}
          >
            <Jss(Flexbox)
              direction="col"
              display="flex"
            >
              <Jss(Flexbox)
                flex="1"
              >
                <Jss(Flexbox)
                  alignItems="start"
                  className="top"
                  display="flex"
                  justifyContent="end"
                >
                  <div
                    className="closeIcon"
                  >
                    <Jss(Icon)
                      colors={
                        Array [
                          "#222222",
                        ]
                      }
                      height={24}
                      iconName="close"
                      onClick={[MockFunction]}
                      transition="0.3s all"
                      width={24}
                    />
                  </div>
                </Jss(Flexbox)>
                <div
                  className="content"
                />
              </Jss(Flexbox)>
            </Jss(Flexbox)>
          </Jss(Card)>
        </div>,
        "className": "cardWrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Jss(Card)
            className="card md cardHide"
            shadow={5}
          >
            <Jss(Flexbox)
              direction="col"
              display="flex"
            >
              <Jss(Flexbox)
                flex="1"
              >
                <Jss(Flexbox)
                  alignItems="start"
                  className="top"
                  display="flex"
                  justifyContent="end"
                >
                  <div
                    className="closeIcon"
                  >
                    <Jss(Icon)
                      colors={
                        Array [
                          "#222222",
                        ]
                      }
                      height={24}
                      iconName="close"
                      onClick={[MockFunction]}
                      transition="0.3s all"
                      width={24}
                    />
                  </div>
                </Jss(Flexbox)>
                <div
                  className="content"
                />
              </Jss(Flexbox)>
            </Jss(Flexbox)>
          </Jss(Card)>,
          "className": "cardBlock",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Jss(Flexbox)
              direction="col"
              display="flex"
            >
              <Jss(Flexbox)
                flex="1"
              >
                <Jss(Flexbox)
                  alignItems="start"
                  className="top"
                  display="flex"
                  justifyContent="end"
                >
                  <div
                    className="closeIcon"
                  >
                    <Jss(Icon)
                      colors={
                        Array [
                          "#222222",
                        ]
                      }
                      height={24}
                      iconName="close"
                      onClick={[MockFunction]}
                      transition="0.3s all"
                      width={24}
                    />
                  </div>
                </Jss(Flexbox)>
                <div
                  className="content"
                />
              </Jss(Flexbox)>
            </Jss(Flexbox)>,
            "className": "card md cardHide",
            "shadow": 5,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                undefined,
                <Jss(Flexbox)
                  flex="1"
                >
                  <Jss(Flexbox)
                    alignItems="start"
                    className="top"
                    display="flex"
                    justifyContent="end"
                  >
                    <div
                      className="closeIcon"
                    >
                      <Jss(Icon)
                        colors={
                          Array [
                            "#222222",
                          ]
                        }
                        height={24}
                        iconName="close"
                        onClick={[MockFunction]}
                        transition="0.3s all"
                        width={24}
                      />
                    </div>
                  </Jss(Flexbox)>
                  <div
                    className="content"
                  />
                </Jss(Flexbox)>,
              ],
              "direction": "col",
              "display": "flex",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Jss(Flexbox)
                      alignItems="start"
                      className="top"
                      display="flex"
                      justifyContent="end"
                    >
                      <div
                        className="closeIcon"
                      >
                        <Jss(Icon)
                          colors={
                            Array [
                              "#222222",
                            ]
                          }
                          height={24}
                          iconName="close"
                          onClick={[MockFunction]}
                          transition="0.3s all"
                          width={24}
                        />
                      </div>
                    </Jss(Flexbox)>,
                    <div
                      className="content"
                    />,
                    undefined,
                  ],
                  "flex": "1",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "alignItems": "start",
                      "children": Array [
                        undefined,
                        <div
                          className="closeIcon"
                        >
                          <Jss(Icon)
                            colors={
                              Array [
                                "#222222",
                              ]
                            }
                            height={24}
                            iconName="close"
                            onClick={[MockFunction]}
                            transition="0.3s all"
                            width={24}
                          />
                        </div>,
                      ],
                      "className": "top",
                      "display": "flex",
                      "justifyContent": "end",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Jss(Icon)
                            colors={
                              Array [
                                "#222222",
                              ]
                            }
                            height={24}
                            iconName="close"
                            onClick={[MockFunction]}
                            transition="0.3s all"
                            width={24}
                          />,
                          "className": "closeIcon",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "colors": Array [
                              "#222222",
                            ],
                            "height": 24,
                            "iconName": "close",
                            "onClick": [MockFunction],
                            "transition": "0.3s all",
                            "width": 24,
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": "div",
                      },
                    ],
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "content",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  undefined,
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cardWrapper"
        >
          <div
            className="cardBlock"
            onClick={[Function]}
          >
            <Jss(Card)
              className="card md cardHide"
              shadow={5}
            >
              <Jss(Flexbox)
                direction="col"
                display="flex"
              >
                <Jss(Flexbox)
                  flex="1"
                >
                  <Jss(Flexbox)
                    alignItems="start"
                    className="top"
                    display="flex"
                    justifyContent="end"
                  >
                    <div
                      className="closeIcon"
                    >
                      <Jss(Icon)
                        colors={
                          Array [
                            "#222222",
                          ]
                        }
                        height={24}
                        iconName="close"
                        onClick={[MockFunction]}
                        transition="0.3s all"
                        width={24}
                      />
                    </div>
                  </Jss(Flexbox)>
                  <div
                    className="content"
                  />
                </Jss(Flexbox)>
              </Jss(Flexbox)>
            </Jss(Card)>
          </div>
        </div>,
        "className": "overlay overlayHide",
        "onClick": [MockFunction],
        "onTransitionEnd": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="cardBlock"
            onClick={[Function]}
          >
            <Jss(Card)
              className="card md cardHide"
              shadow={5}
            >
              <Jss(Flexbox)
                direction="col"
                display="flex"
              >
                <Jss(Flexbox)
                  flex="1"
                >
                  <Jss(Flexbox)
                    alignItems="start"
                    className="top"
                    display="flex"
                    justifyContent="end"
                  >
                    <div
                      className="closeIcon"
                    >
                      <Jss(Icon)
                        colors={
                          Array [
                            "#222222",
                          ]
                        }
                        height={24}
                        iconName="close"
                        onClick={[MockFunction]}
                        transition="0.3s all"
                        width={24}
                      />
                    </div>
                  </Jss(Flexbox)>
                  <div
                    className="content"
                  />
                </Jss(Flexbox)>
              </Jss(Flexbox)>
            </Jss(Card)>
          </div>,
          "className": "cardWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Jss(Card)
              className="card md cardHide"
              shadow={5}
            >
              <Jss(Flexbox)
                direction="col"
                display="flex"
              >
                <Jss(Flexbox)
                  flex="1"
                >
                  <Jss(Flexbox)
                    alignItems="start"
                    className="top"
                    display="flex"
                    justifyContent="end"
                  >
                    <div
                      className="closeIcon"
                    >
                      <Jss(Icon)
                        colors={
                          Array [
                            "#222222",
                          ]
                        }
                        height={24}
                        iconName="close"
                        onClick={[MockFunction]}
                        transition="0.3s all"
                        width={24}
                      />
                    </div>
                  </Jss(Flexbox)>
                  <div
                    className="content"
                  />
                </Jss(Flexbox)>
              </Jss(Flexbox)>
            </Jss(Card)>,
            "className": "cardBlock",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Jss(Flexbox)
                direction="col"
                display="flex"
              >
                <Jss(Flexbox)
                  flex="1"
                >
                  <Jss(Flexbox)
                    alignItems="start"
                    className="top"
                    display="flex"
                    justifyContent="end"
                  >
                    <div
                      className="closeIcon"
                    >
                      <Jss(Icon)
                        colors={
                          Array [
                            "#222222",
                          ]
                        }
                        height={24}
                        iconName="close"
                        onClick={[MockFunction]}
                        transition="0.3s all"
                        width={24}
                      />
                    </div>
                  </Jss(Flexbox)>
                  <div
                    className="content"
                  />
                </Jss(Flexbox)>
              </Jss(Flexbox)>,
              "className": "card md cardHide",
              "shadow": 5,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  undefined,
                  <Jss(Flexbox)
                    flex="1"
                  >
                    <Jss(Flexbox)
                      alignItems="start"
                      className="top"
                      display="flex"
                      justifyContent="end"
                    >
                      <div
                        className="closeIcon"
                      >
                        <Jss(Icon)
                          colors={
                            Array [
                              "#222222",
                            ]
                          }
                          height={24}
                          iconName="close"
                          onClick={[MockFunction]}
                          transition="0.3s all"
                          width={24}
                        />
                      </div>
                    </Jss(Flexbox)>
                    <div
                      className="content"
                    />
                  </Jss(Flexbox)>,
                ],
                "direction": "col",
                "display": "flex",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Jss(Flexbox)
                        alignItems="start"
                        className="top"
                        display="flex"
                        justifyContent="end"
                      >
                        <div
                          className="closeIcon"
                        >
                          <Jss(Icon)
                            colors={
                              Array [
                                "#222222",
                              ]
                            }
                            height={24}
                            iconName="close"
                            onClick={[MockFunction]}
                            transition="0.3s all"
                            width={24}
                          />
                        </div>
                      </Jss(Flexbox)>,
                      <div
                        className="content"
                      />,
                      undefined,
                    ],
                    "flex": "1",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "alignItems": "start",
                        "children": Array [
                          undefined,
                          <div
                            className="closeIcon"
                          >
                            <Jss(Icon)
                              colors={
                                Array [
                                  "#222222",
                                ]
                              }
                              height={24}
                              iconName="close"
                              onClick={[MockFunction]}
                              transition="0.3s all"
                              width={24}
                            />
                          </div>,
                        ],
                        "className": "top",
                        "display": "flex",
                        "justifyContent": "end",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": <Jss(Icon)
                              colors={
                                Array [
                                  "#222222",
                                ]
                              }
                              height={24}
                              iconName="close"
                              onClick={[MockFunction]}
                              transition="0.3s all"
                              width={24}
                            />,
                            "className": "closeIcon",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "class",
                            "props": Object {
                              "colors": Array [
                                "#222222",
                              ],
                              "height": 24,
                              "iconName": "close",
                              "onClick": [MockFunction],
                              "transition": "0.3s all",
                              "width": 24,
                            },
                            "ref": null,
                            "rendered": null,
                            "type": [Function],
                          },
                          "type": "div",
                        },
                      ],
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "content",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "div",
                    },
                    undefined,
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Modal styles matches the snapshot 1`] = `
Object {
  "@global": Object {
    "body": Object {
      "overflow": "hidden",
    },
  },
  "@keyframes modalFadeDown": Object {
    "from": Object {
      "marginTop": -64,
      "opacity": 0,
    },
    "to": Object {
      "marginTop": 0,
      "opacity": 1,
    },
  },
  "@keyframes modalFadeIn": Object {
    "from": Object {
      "opacity": 0,
    },
    "to": Object {
      "opacity": 1,
    },
  },
  "bottom": Object {
    "marginTop": 8,
    "textAlign": "right",
  },
  "card": Object {
    "@media screen and (min-width:768px)": Object {
      "padding": 32,
    },
    "cursor": "auto",
    "maxWidth": "100%",
    "padding": 16,
    "transition": "0.3s opacity, 0.3s margin-top",
  },
  "cardBlock": Object {
    "margin": 12,
  },
  "cardHide": Object {
    "marginTop": -64,
    "opacity": 0,
  },
  "cardShow": Object {
    "animation": "0.3s modalFadeDown ease-out",
  },
  "cardWrapper": Object {
    "maxHeight": "100vh",
    "maxWidth": "100vw",
    "position": "relative",
  },
  "closeIcon": Object {
    "alignItems": "center",
    "display": "flex",
    "height": 32,
  },
  "closePosition": Object {
    "@media screen and (min-width:768px)": Object {
      "right": 32,
      "top": 32,
    },
    "position": "absolute",
    "right": 16,
    "top": 16,
  },
  "content": Object {
    "&::-webkit-scrollbar": Object {
      "width": 6,
    },
    "&::-webkit-scrollbar-thumb": Object {
      "background": "#dddddd",
      "borderRadius": 3,
    },
    "@media screen and (min-width:768px)": Object {
      "marginLeft": -32,
      "marginRight": -32,
      "paddingLeft": 32,
      "paddingRight": 32,
    },
    "marginLeft": -16,
    "marginRight": -16,
    "paddingLeft": 16,
    "paddingRight": 16,
    "transition": "0.1s height",
  },
  "imgLeft": Object {
    "@media screen and (min-width:768px)": Object {
      "marginBottom": -32,
      "marginLeft": -32,
      "marginRight": 32,
      "marginTop": -32,
    },
    "background": [Function],
    "borderRadius": Array [
      6,
      0,
      0,
      6,
    ],
    "marginBottom": -16,
    "marginLeft": -16,
    "marginRight": 16,
    "marginTop": -16,
    "width": 200,
  },
  "imgTop": Object {
    "@media screen and (min-width:768px)": Object {
      "marginBottom": 32,
      "marginLeft": -32,
      "marginRight": -32,
      "marginTop": -32,
    },
    "background": [Function],
    "borderRadius": Array [
      6,
      6,
      0,
      0,
    ],
    "height": 128,
    "marginBottom": 16,
    "marginLeft": -16,
    "marginRight": -16,
    "marginTop": -16,
    "position": "relative",
  },
  "lg": Object {
    "width": 820,
  },
  "mainBtn": Object {
    "marginLeft": 24,
    "marginTop": 16,
  },
  "md": Object {
    "@media screen and (max-width:767px)": Object {
      "maxWidth": "100%",
      "width": 322,
    },
    "width": 520,
  },
  "noClose": Object {
    "cursor": "default",
  },
  "overlay": Object {
    "alignItems": "center",
    "background": "rgba(0, 0, 5, 0.85)",
    "cursor": "pointer",
    "display": "flex",
    "height": "100%",
    "justifyContent": "center",
    "left": 0,
    "overflow": "auto",
    "position": "fixed",
    "top": 0,
    "transition": "0.3s all",
    "width": "100%",
    "zIndex": 1000,
  },
  "overlayHide": Object {
    "opacity": 0,
  },
  "overlayShow": Object {
    "animation": "0.3s modalFadeIn linear",
  },
  "secBtn": Object {
    "marginTop": 16,
  },
  "sm": Object {
    "maxWidth": "100%",
    "width": 322,
  },
  "top": Object {
    "marginBottom": 16,
  },
  "xl": Object {
    "width": 1024,
  },
}
`;

exports[`ModalPortal matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalPortal
    container={<div />}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
